src/nrnoc/extcelln.c

```diff
@@ -193,13 +202,97 @@ hoc_execerror("Extracellular mechanism only works with fixed step methods and da
 	}
 }
 
+void extnode_free_elements(Extnode* nde) {
+  if (nde->v) {
+    free(nde->v); /* along with _a and _b */
+    free(nde->_d); /* along with _rhs, _a_matelm, _b_matelm, _x12, and _x21 */
+    nde->v = NULL;
+    nde->_a = NULL;
+    nde->_b = NULL;
+    nde->_d = NULL;
+    nde->_rhs = NULL;
+    nde->_a_matelm = NULL;
+    nde->_b_matelm = NULL;
+    nde->_x12 = NULL;
+    nde->_x21 = NULL;
+  }
+}
+
+static void check_if_extracellular_in_use() {
+  hoc_Item* qsec;
+  ITERATE(qsec, section_list) {
+    Section* sec = hocSEC(qsec);
+    if (sec->pnode[0]->extnode) {
+      hoc_execerror("Cannot change nlayer_extracellular when instances exist", NULL);
+    }
+  }
+}
+
+static void update_existing_extnode(int old_nlayer) {
+  /* there aren't any because of check_if_extracellular_in_use() */
+}
+
+static void update_extracellular_reg(int old_nlayer) {
+  /* update hoc_symlist for arayinfo->sub[0] and u.rng.index. */
+  int i, ix;
+  Symbol* ecell = hoc_table_lookup("extracellular", hoc_built_in_symlist);
+  assert(ecell);
+  assert(ecell->type == MECHANISM);
+  ix = 0;
+  for (i=0; i < ecell->s_varn; ++i) {
+    Symbol* s = ecell->u.ppsym[i];
+    if (s->type == RANGEVAR) {
+      Arrayinfo* a = s->arayinfo;
+      s->u.rng.index = ix;
+      if (a && a->nsub == 1) {
+        assert(a->sub[0] == old_nlayer);
+        a->sub[0] = nlayer;
+        ix += nlayer;
+      }else{
+        ix += 1;
+      }
+    }
+  }
+}
+
+void nlayer_extracellular() {
+  if (ifarg(1)) {
+    int old = nlayer;
+    nrn_nlayer_extracellular = (int)chkarg(1, 1., 1000.);
+    if (nlayer == old) { return; }
+
+    check_if_extracellular_in_use();
+    nrn_delete_prop_pool(EXTRACELL);
+    update_extracellular_reg(old);
+    update_existing_extnode(old);
```

Comment:
bit got confused here. I thought if user has now passed new value for layers then here the argument should be `nrn_nlayer_extracellular` instead of `old`.

Reply:
nlayer has already been changed so those callees know the new value and may need to be informed of the previous value.